### ISSUE TITLE
When "agg_trade" is running, I can't terminate the program using Ctrl+C. Is there any solution to this?

### ISSUE THREAD

**Issue by IronEaterBeast** [binance/binance-connector-python#345]
from binance.websocket.spot.websocket_stream import SpotWebsocketStreamClient

def on_message(_, _message):
    print(_message)

my_client = SpotWebsocketStreamClient(on_message=on_message)
my_client.agg_trade(symbol="btcusdt")

**Comment by alplabin** [comment]
Have you considered closing the connection using the method `my_client.stop()` like in the [example](https://github.com/binance/binance-connector-python/blob/master/examples/websocket/spot/websocket_stream/agg_trade.py#L30)?

**Comment by IronEaterBeast** [comment]
> Have you considered closing the connection using the method `my_client.stop()` like in the [example](https://github.com/binance/binance-connector-python/blob/master/examples/websocket/spot/websocket_stream/agg_trade.py#L30)?

Yes, I know about my_client.stop(), but what I want is for the program to keep running until I manually close it, not to stop automatically after a set time. How can I make it so that I can decide whether it should close at any time after the program starts running?

**Comment by alplabin** [comment]
What exactly happens when you Ctrl+c? Which os are you using?

**Comment by IronEaterBeast** [comment]
> What exactly happens when you Ctrl+c? Which os are you using?

I run the program in the terminal of VSCode on Windows. After pressing Ctrl+c, the program continues its previous running state, constantly fetching the latest data from a remote server and printing the data on the screen. This terminal cannot respond to any other operations.

**Comment by Lokesh2003GN** [comment]
You can wrap your program in a try-except block to catch the KeyboardInterrupt and close the connection cleanly when Ctrl+C is pressed.

import logging
from binance.lib.utils import config_logging
from binance.websocket.spot.websocket_stream import SpotWebsocketStreamClient

config_logging(logging, logging.DEBUG)

def message_handler(_, message):
    logging.info(message)

# Initialize the client
my_client = SpotWebsocketStreamClient(on_message=message_handler, is_combined=True)

try:
    # Subscribe to a single symbol stream
    my_client.agg_trade(symbol="bnbusdt")
    print("Streaming started. Press Ctrl+C to terminate.")
    
    # Keep the program running
    while True:
        pass  

except KeyboardInterrupt:
    print("\nCtrl+C pressed. Stopping the client...")
    my_client.stop()

**Comment by alplabin** [comment]
Hi @IronEaterBeast , I'm closing this issue as we published a brand new modularised version of the Python connector. Feel free to start using it and open a new issue if you encounter any problems.
